FakeHttpService.Tests/Given_I_want_to_get_fake_responses_in_a_test.cs

```diff
@@ -15,28 +17,159 @@ namespace FakeHttpService.Tests
 {
     public class Given_I_want_to_get_fake_responses_in_a_test : IDisposable
     {
-        private readonly IDisposable disposables;
+        private readonly IDisposable _disposables;
 
         public Given_I_want_to_get_fake_responses_in_a_test(ITestOutputHelper output)
         {
-            disposables = LogEvents.Subscribe(e => output.WriteLine(e.ToLogString()));
+            _disposables = LogEvents.Subscribe(e => output.WriteLine(e.ToLogString()));
         }
 
-        public void Dispose() => disposables.Dispose();
+        public void Dispose() => _disposables.Dispose();
 
         [Fact]
         public async Task When_requesting_a_registered_Uri_Then_the_expected_response_is_returned()
         {
             using (var fakeService = new FakeHttpService()
                 .OnRequest(r => true)
-                .RespondWith(async r => {}))
+                .RespondWith(async r => { await Task.Yield(); }))
             {
                 var response = await new HttpClient().GetAsync(fakeService.BaseAddress);
 
                 response.EnsureSuccessStatusCode();
             }
         }
 
+        [Fact]
+        public async Task When_filtering_Uri_Then_the_expected_response_is_returned()
+        {
+            using (var fakeService = new FakeHttpService()
+                .OnRequest()
+                .WhereUri(uri => uri.ToString().EndsWith("customapicall"))
+                .Then()
+                .RespondWith(async r => {
+                    r.StatusCode = 200;
+                    await Task.Yield();
+                }))
+            {
+                var response = await new HttpClient().GetAsync( new Uri(fakeService.BaseAddress, "/customapicall"));
+
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        [Fact]
+        public async Task When_filtering_method_Then_the_expected_response_is_returned()
+        {
+            using (var fakeService = new FakeHttpService()
+                .OnRequest()
+                .WhereMehtod(method => method == "POST")
+                .Then()
+                .RespondWith(async r =>
+                {
+                    r.StatusCode = 200;
+                    await Task.Yield();
+                })
+            .OnRequest(_ => true).RespondWith(async r =>
+                {
```

Comment:
This feels like a useful config setting on the service level. "FailOnUnexpectedRequest"?

Reply:
indeed